src/Collections/TerminusCollection.php

```diff
@@ -91,6 +94,22 @@ public function all()
     public function fetch()
     {
         foreach ($this->getData() as $id => $model_data) {
+            if (!is_object($model_data)) {
+                // For some reason I can't replicate, occasionally $model_data is just a string here.
+                $trace = debug_backtrace();
+                $error_message = "Fetch failed {file}:{line} model_data expected as object but returned as {type}.";
+                $context = [
+                    'file' => $trace[0]['file'],
+                    'line' => $trace[0]['line'],
+                    'type' => gettype($model_data),
+                ];
+                if (is_string($model_data)) {
+                    $error_message .= " String value: {string}";
```

Comment:
let's try this again...

```suggestion
                if (is_scalar($model_data)) {
```
@greg-1-anderson how do you feel about is_scalar() here? I tested locally with php -r and it does implicit string conversion for integer, float, and boolean values in the string functions on line 108. float and integer behave predictably. boolean becomes "1" for true and "" for false.

Reply:
it did it again!